pkg/cluster/spec/tici_meta.go

```diff
@@ -0,0 +1,274 @@
+func (s *TiCIMetaSpec) IgnoreMonitorAgent() bool {
+	return s.IgnoreExporter
+}
+
+// TiCIMetaComponent represents TiCI Meta Server component.
+type TiCIMetaComponent struct{ Topology *Specification }
+
+// Name implements Component interface.
+func (c *TiCIMetaComponent) Name() string {
+	return ComponentTiCIMeta
+}
+
+// Role implements Component interface.
+func (c *TiCIMetaComponent) Role() string {
+	return ComponentTiCIMeta
+}
+
+// Source implements Component interface.
+func (c *TiCIMetaComponent) Source() string {
+	source := c.Topology.ComponentSources.TiCI
+	if source != "" {
+		return source
+	}
+	return ComponentTiCI
+}
+
+// CalculateVersion implements the Component interface
+func (c *TiCIMetaComponent) CalculateVersion(clusterVersion string) string {
+	version := c.Topology.ComponentVersions.TiCIMeta
+	if version == "" {
+		version = clusterVersion
+	}
+	return version
+}
+
+// SetVersion implements Component interface.
+func (c *TiCIMetaComponent) SetVersion(version string) {
+	c.Topology.ComponentVersions.TiCIMeta = version
+}
+
+// Instances implements Component interface.
+func (c *TiCIMetaComponent) Instances() []Instance {
+	ins := make([]Instance, 0, len(c.Topology.TiCIMetaServers))
+	for _, s := range c.Topology.TiCIMetaServers {
+		ins = append(ins, &TiCIMetaInstance{BaseInstance{
+			InstanceSpec: s,
+			Name:         c.Name(),
+			Host:         s.Host,
+			ManageHost:   s.ManageHost,
+			ListenHost:   utils.Ternary(s.ListenHost != "", s.ListenHost, c.Topology.BaseTopo().GlobalOptions.ListenHost).(string),
+			Port:         s.Port,
+			SSHP:         s.SSHPort,
+			Source:       s.Source,
+			NumaNode:     s.NumaNode,
+			NumaCores:    s.NumaCores,
+
+			Ports: []int{
+				s.Port,
+				s.StatusPort,
+			},
+			Dirs: []string{
+				s.DeployDir,
+			},
+			StatusFn: func(_ context.Context, timeout time.Duration, tlsCfg *tls.Config, _ ...string) string {
+				return statusByHost(s.GetManageHost(), s.StatusPort, "/status", timeout, tlsCfg)
+			},
+			UptimeFn: func(_ context.Context, timeout time.Duration, tlsCfg *tls.Config) time.Duration {
+				return UptimeByHost(s.GetManageHost(), s.StatusPort, timeout, tlsCfg)
+			},
+			Component: c,
+		}, c.Topology})
+	}
+	return ins
+}
+
+// TiCIMetaInstance represent the TiCI Meta Server instance
+type TiCIMetaInstance struct {
+	BaseInstance
+	topo Topology
+}
+
+// InitConfig implement Instance interface
+func (i *TiCIMetaInstance) InitConfig(
+	ctx context.Context,
+	e ctxt.Executor,
+	clusterName,
+	clusterVersion,
+	deployUser string,
+	paths meta.DirPaths,
+) error {
+	topo := i.topo.(*Specification)
+	if err := i.BaseInstance.InitConfig(ctx, e, topo.GlobalOptions, deployUser, paths); err != nil {
+		return err
+	}
+
+	spec := i.InstanceSpec.(*TiCIMetaSpec)
+
+	// Generate PD endpoints for TiCI Meta
+	// currently TiCI Meta only supports single PD
+	pd := utils.JoinHostPort(topo.PDServers[0].Host, topo.PDServers[0].ClientPort)
```

Comment:
Please create an issue for supporting multiple PD address in TiCI meta and TiCI worker in case we forget it before TiCI GA